ios/RNSScreenStackComponentView.mm

```diff
@@ -265,11 +286,145 @@ - (void)dismissOnReload
 
 #pragma mark methods connected to transitioning
 
+- (void)cancelTouchesInParent
+{
+  // cancel touches in parent, this is needed to cancel RN touch events. For example when Touchable
+  // item is close to an edge and we start pulling from edge we want the Touchable to be cancelled.
+  // Without the below code the Touchable will remain active (highlighted) for the duration of back
+  // gesture and onPress may fire when we release the finger.
+  UIView *parent = _controller.view;
+  while (parent != nil && ![parent respondsToSelector:@selector(touchHandler)])
+    parent = parent.superview;
+
+  if (parent != nil) {
+    RCTSurfaceTouchHandler *touchHandler = [parent performSelector:@selector(touchHandler)];
+    [touchHandler setEnabled:NO];
```

Comment:
Is calling those 2 methods the same as this: https://github.com/software-mansion/react-native-screens/blob/844295039e84697ba25341a43f2fbb50e5c7fb85/ios/RNSScreenStack.m#L600 earlier? Did you test this?

Reply:
It should be. The thing is, that in new counterpart of `RCTTouchHandler` -- `RCTSurfaceTouchHandler` the `cancel` method became private, but it has the same impl as before ([old impl](https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/Base/RCTTouchHandler.m#L362-L366), [new impl](https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/React/Fabric/RCTSurfaceTouchHandler.mm#L425-L429)).

PS. Tested -- seems to work properly